File: IAIso-v5.0/core/iaiso-php/src/Coordination/Snapshot.php
<?php

declare(strict_types=1);

namespace IAIso\Coordination;

/** Read-only view of coordinator state. */
final class Snapshot
{
    /** @param array<string,float> $perExecution */
    public function __construct(
        public readonly string $coordinatorId,
        public readonly float $aggregatePressure,
        public readonly CoordinatorLifecycle $lifecycle,
        public readonly int $activeExecutions,
        public readonly array $perExecution,
    ) {
    }
}

File: IAIso-v5.0/core/iaiso-swift/Sources/IAIsoCoordination/Snapshot.swift
import Foundation

/// Read-only view of coordinator state.
public struct Snapshot: Sendable, Equatable {
    public let coordinatorId: String
    public let aggregatePressure: Double
    public let lifecycle: CoordinatorLifecycle
    public let activeExecutions: Int
    public let perExecution: [String: Double]

    public init(
        coordinatorId: String,
        aggregatePressure: Double,
        lifecycle: CoordinatorLifecycle,
        activeExecutions: Int,
        perExecution: [String: Double]
    ) {
        self.coordinatorId = coordinatorId
        self.aggregatePressure = aggregatePressure
        self.lifecycle = lifecycle
        self.activeExecutions = activeExecutions
        self.perExecution = perExecution
    }
}
